### ISSUE TITLE
Property with inproper casing found when checking missingMemberHandling, but not for assigning value.

### ISSUE THREAD

**Issue by sbeaudoin** [JamesNK/Newtonsoft.Json#736]
I have a class with a "Id" property (uppercase "I").  I deserialize a string where "id" have a value of 23 for example (lowercase "i").  My deserialization settings have "MissingMemberHandling=Error".

When I deserialize the string, no error occur and the property Id have the default 0 value, not 23 as expected.

If I rename my property for "id", lowercase "i", the value 23 is correctly assigned.

If I rename my property for "idz", an error occur as expected, indicating "id" property is not found.

I suspect the property presence validation is case insensitive, but the code who find the property to assign the value is case sensitive.

**Comment by JamesNK** [comment]
Works for me

        [Test]
        public void CaseInsensitive()
        {
            string json = @"{""height"":1}";

            DoubleClass c = JsonConvert.DeserializeObject<DoubleClass>(json, new JsonSerializerSettings { MissingMemberHandling = MissingMemberHandling.Error });

            Assert.AreEqual(1d, c.Height);
        }

**Comment by sbeaudoin** [comment]
I traced in your library, The probem is, the call of "contract.Properties.GetClosestMatchProperty(memberName)" in the PopulateObject method return null if the property is "internal set".

Modify the property Height of Doubleclass with an internal set and you will be able to reproduce the problem.

The deserialized object I create from the json must not be modified after it exit my assemby, so I used internal set.
